compiler/rustc_hir_analysis/src/check/compare_method.rs

```diff
@@ -153,13 +153,10 @@ fn compare_predicate_entailment<'tcx>(
     // This node-id should be used for the `body_id` field on each
     // `ObligationCause` (and the `FnCtxt`).
     //
-    // FIXME(@lcnr): remove that after removing `cause.body_id` from
-    // obligations.
-    let impl_m_hir_id = tcx.hir().local_def_id_to_hir_id(impl_m.def_id.expect_local());
     // We sometimes modify the span further down.
```

Comment:
can you instead use `let impl_m_def_id = impl_m.def_id.expect_local()` instead of using `expect_local()` multiple times in this method?